src/common/catalog/catalog-entity-registry.ts

```diff
@@ -19,27 +19,33 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import { action, computed, observable, IComputedValue, IObservableArray } from "mobx";
+import { action, computed, IComputedValue, IObservableArray, makeObservable, observable } from "mobx";
 import { CatalogEntity } from "./catalog-entity";
-import { iter } from "../utils";
 
 export class CatalogEntityRegistry {
-  protected sources = observable.map<string, IComputedValue<CatalogEntity[]>>([], { deep: true });
+  protected sources = observable.map<string, IComputedValue<CatalogEntity[]>>();
 
-  @action addObservableSource(id: string, source: IObservableArray<CatalogEntity>) {
-    this.sources.set(id, computed(() => source));
+  constructor() {
+    makeObservable(this);
   }
 
-  @action addComputedSource(id: string, source: IComputedValue<CatalogEntity[]>) {
-    this.sources.set(id, source);
+  @action
+  addObservableSource(id: string, source: IObservableArray<CatalogEntity> | IComputedValue<CatalogEntity[]>) {
+    if (Array.isArray(source)) {
+      this.sources.set(id, computed(() => source.toJSON()));
+    } else {
+      this.sources.set(id, source);
+    }
   }
 
   @action removeSource(id: string) {
     this.sources.delete(id);
   }
 
   @computed get items(): CatalogEntity[] {
-    return Array.from(iter.flatMap(this.sources.values(), source => source.get()));
+    return Array.from(this.sources.values())
+      .map(source => source.get())
+      .flat();
```

Comment:
Please revert things like this. This just clutters up the diff making it hard to review.

Reply:
I wasn't sure previous code was working with mobx-6. So, I'll prefer to keep it as is for now.
Moreover I think form is more readable than previous and with less extra apis/helpers involved).

Reply:
I completely disagree, if you can show that it wasn't working then sure go ahead. But from all the docs of how JS's `Array.from()` works. The previous code was correct.

Reply:
Yeah, maybe your version is correct. But I prefer mine for more simplicity.

Reply:
How is it more simple? Also a migration PR is not the place for this sort of thing.

Reply:
It simple because:
- uses only native JS apis
- don't use generators/iterators (less complexity imho)

Let's ask from @lensapp/lens-maintainers which form they think more simple.
Anyway, I can revert these just to avoid tedious arguing with you.

Reply:
Reverted.